Journal 2/Journal/EntriesTableViewController.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  EntriesTableViewController.swift
+//  Project
+//
+//  Created by Jonathan Ferrer on 6/3/19.
+//  Copyright © 2019 Jonathan Ferrer. All rights reserved.
+//
+
+import UIKit
+import CoreData
+
+
+
+class EntriesTableViewController: UITableViewController, NSFetchedResultsControllerDelegate{
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+    }
+    @IBAction func refresh(_ sender: UIRefreshControl) {
+
+        entryController.fetchEntriesFromServer() { (_) in
+            DispatchQueue.main.async {
+                self.refreshControl?.endRefreshing()
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    // MARK: - Table view data source
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        tableView.reloadData()
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return fetchedResultsdController.sections?.count ?? 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+
+        return fetchedResultsdController.sections?[section].numberOfObjects ?? 0
+    }
+
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "EntryCell", for: indexPath) as! EntryTableViewCell
+
+        // Configure the cell...
+        let entry = fetchedResultsdController.object(at: indexPath)
+        cell.entry = entry
+
+
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        guard let sectionInfo = fetchedResultsdController.sections?[section] else {
+            return nil
+        }
+        return sectionInfo.name
+    }
+
+    // Override to support editing the table view.
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+
+            let entry = fetchedResultsdController.object(at: indexPath)
+            let moc = CoreDataStack.shared.mainContext
```

Comment:
You no longer need:
 ```
let moc = CoreDataStack.shared.mainContext
	            moc.delete(entry)
	           
	            do {
	                try moc.save()
	                tableView.reloadData()
	            } catch {
	                moc.reset()
	                NSLog("Error deleting contrext: \(error)")
	            }
```
	
Because you have the delete function from the model controller so you can just call it like this:

`entryController.deleteEntry(entry: entry)`